EssentialFeed/EssentialFeediOS/Image Comment UI/Controllers/ImageCommentsViewController.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  ImageCommentsViewController.swift
+//  EssentialFeediOS
+//
+//  Created by Alok Subedi on 06/02/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import EssentialFeed
+
+public protocol ImageCommentsViewControllerDelegate {
+	func didRequestImageCommentsRefresh()
+}
+
+public class ImageCommentsViewController: UITableViewController, ImageCommentsView, ImageCommentsLoadingView, ImageCommentsErrorView {
+	@IBOutlet private(set) public var errorView: ErrorView?
+	
+	public var delegate: ImageCommentsViewControllerDelegate?
+	private var imageComments = [PresentableImageComment]() {
+		didSet {
+			tableView.reloadData()
+		}
+	}
+	
+	public override func viewDidLoad() {
+		super.viewDidLoad()
+		
+		load()
+	}
+	
+	public override func viewDidLayoutSubviews() {
+		super.viewDidLayoutSubviews()
+		
+		tableView.sizeTableHeaderToFit()
+	}
+	
+	@IBAction private func load() {
+		delegate?.didRequestImageCommentsRefresh()
+	}
+	
+	public func display(_ viewModel: ImageCommentsViewModel) {
+		self.imageComments = viewModel.comments
+	}
+	
+	public func display(_ viewModel: ImageCommentsLoadingViewModel) {
+		refreshControl?.update(isRefreshing: viewModel.isLoading)
+	}
+	
+	public func display(_ viewModel: ImageCommentsErrorViewModel) {
+		if let errorMessage = viewModel.message {
+			errorView?.message = errorMessage
+		} else {
+			errorView?.message = nil
+		}
```

Comment:
The if/else is unnecessary. The result is the same as:

```suggestion
		errorView?.message = errorMessage
```